### DISCUSSION TITLE
How to show/hide layers of styleURL on the client

### DISCUSSION THREAD

**Discussion by CarlosAlbertoTI** [rnmapbox/maps#2961] [Q&A]
Hey guys, I would like to know if there is a way to dynamically hide/show layers that I have implemented or set up on Mapbox Studio on the client side. Here's an example scenario:

1) Let's say I have added a layer of houses with points on Mapbox Studio.
2) In my app, if a local layer of rivers is set to be displayed on the map, I want the house layers to be disabled.

I came across a similar post, but it is specifically related to [react-map-gl](https://i.stack.imgur.com/frn9U.png). Here's the link:
https://stackoverflow.com/questions/74102313/how-to-show-hide-layers-in-mapbox-on-the-client

**Comment by Gnative** [comment]
I think you should be able to do this with [setSourceVisibility](https://github.com/rnmapbox/maps/blob/main/docs/MapView.md#setsourcevisibilityvisible-sourceid-sourcelayerid)

There is also an example here - https://github.com/rnmapbox/maps/blob/main/example/src/examples/Map/SourceLayerVisibility.js
